mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -40,7 +41,8 @@ def __init__(self, tunables: TunableGroups, service: Optional[Service], config:
         self._tunables = tunables
         self._service = service
         self._iter = 1
-        self._max_iter = int(self._config.pop('max_iterations', 10))
+        self._use_defaults: bool = bool(strtobool(str.lower(self._config.pop('use_defaults', "True"))))
```

Comment:
It's a bit tricky. We use JSON5 parser, so the config can have `"use_defaults": true` and the value will show up as `bool` in `self._config`. OTOH, the user can override that value in the command line by writing `--use_defaults True`, and then it will be a string. We have to check for both cases here.

Reply:
Exactly.  So in this case we convert both the string and the bool to a string, and then parse it back to a bool.  Not awesome, but more robust.